lib/include/arm_controller/arm_controller.h

```diff
@@ -29,279 +18,605 @@
+         * 
+         * param      Pose* pose, and three floats indicating the x-y-z-w coordinates of an orientation
+         *             
+         * return     N/A
+         */
+         
+        static void setOrientation(geometry_msgs::Pose * pose, float x, float y, float z, float w);
+ 
+        /*
+         * sets the joint names of a JointCommand
+         * 
+         * param      JointCommand * joint_cmd, and a string (left/right) indicating which arm is
+         *            being moved
+         *             
+         * return     N/A
+         */
+
+        static void setNames(baxter_core_msgs::JointCommand * joint_cmd, std::string limb);
+
+        /*
+         * converts an integer to a string
+         * 
+         * param      integer to be converted
+         *             
+         * return     converted string
+         */
+ 
+        static std::string intToString( const int a );
+};
+
+// Class initializes overhead functions necessary to start a thread from within a class, 
+// and overhead ROS features: subscriber/publishers, services, callback functions etc.
+class ROSThreadClass
+{
+    public:
+        ROSThreadClass(std::string limb);
+        virtual ~ROSThreadClass();
+
+        pthread_mutex_t _mutex_img;
+        pthread_mutex_t _mutex_pos;    
+        // pthread_mutex_t _mutex_rng;
+
+        pthread_mutexattr_t _attr_img;
+        pthread_mutexattr_t _attr_pos;
+
+        /*
+         * starts thread that executes the internal thread entry function
+         * 
+         * param      N/A
+         * 
+         * return     true if thread was successfully launched; false otherwise
+         */        
+
+        bool StartInternalThread();
+
+        /*
+         * prevents any following code from being executed before thread is exited
+         * 
+         * param      N/A
+         * 
+         * return     true if thread was successfully launched; false otherwise
+         */      
+
+        void WaitForInternalThreadToExit();
+
+        /*
+         * callback function that sets the current pose to the pose received from 
+         * the endpoint state topic
+         * 
+         * param      N/A
+         * 
+         * return     N/A
+         */
+
+        void endpointCallback(const baxter_core_msgs::EndpointState& msg);
+
+        /*
+         * infrared sensor callback function that sets the current range to the range received
+         * from the left hand range state topic
+         * 
+         * param      ImageConstPtr is equal to 'typedef boost::shared_ptr< ::sensor_msgs::Image const>'
+         * 
+         * return     N/A
+         */
+
+        void IRCallback(const sensor_msgs::RangeConstPtr& msg);
+
+        /*
+         * image callback function that displays the image stream from the hand camera 
+         * 
+         * param      ImageConstPtr is equal to 'typedef boost::shared_ptr< ::sensor_msgs::Image const>'
+         * 
+         * return     N/A
+         */
+
```

Comment:
Remove this space between the documentation and the method because doxygen does not like it if I recall correctly